apps/web/lib/getMainDomainOrgRedirect.ts

```diff
@@ -0,0 +1,34 @@
+import type { IncomingMessage } from "http";
+
+import { WEBAPP_URL } from "@calcom/lib/constants";
+import logger from "@calcom/lib/logger";
+
+const log = logger.getSubLogger({ prefix: ["lib", "getMainDomainOrgRedirect"] });
+export const getMainDomainOrgRedirect = (req: IncomingMessage | undefined) => {
+  if (!req) {
+    return null;
+  }
+
+  const protocol = WEBAPP_URL.startsWith("https://") ? "https" : "http";
+  const originalUrl = new URL(req.url!, `${protocol}://${req.headers.host}`);
+
+  const domains = originalUrl.hostname.split(".");
+
+  if (domains.length <= 2) {
+    return null;
+  }
+  log.debug(`Looking for redirect from`, originalUrl.toString());
+
+  const newHostName = `${domains[0]}.${domains.at(-1)}`;
+  originalUrl.hostname = newHostName;
```

Comment:
If url is acme.cal.com, that doesn't necessarily mean their domain has to be acme.com. Can be anything really, imo we should take url input from them and redirect to that url. Similar to how we do it for successful booking redirect.

@PeerRich what do you think about this?

Reply:
In case, we take input from user, we need to validate against our process.env.ALLOWED_HOSTS.

And in any case whether we take input or just assume `acme.cal.com` to be redirected to `acme.com`, we need to append ALLOWED_HOSTS with '"com","dev","community".

This is because there is some validation happening in orgDomain.ts
https://github.com/vijayraghav-io/cal.com/blob/3a46375fed575dc9a35bea2c534044f3a1f9fb2b/packages/features/ee/organizations/lib/orgDomains.ts#L35

if we just redirect without ALLOWED_HOSTS being appended with "com", this validation will fail, and org slug is not identified correctly.

@PeerRich , can you please share your thoughts on this?
Also, please confirm if we need to accept input from user for redirect.

Reply:
Since we are redirecting hostname itself, we should not take input from user, this leads to failed validation against ALLOWED_HOSTNAMES and org slug will not identified correctly.

Reply:
@Amit91848 @vijayraghav-io we need to use the verified "autoInviteURL" for this, @hariombalhara knows

Reply:
> we should take url input from them and redirect to that url. Similar to how we do it for successful booking redirect.

we should not take url input from them, cause this can lead to an abuse vector (adidas.cal.com => adidas.phishing-website.com) instead we should redirect to their verified Org domain (the one they use for all auto-invites for emails)

it's also one less input

Reply:
We have `organizationSettings.orgAutoAcceptEmail` that stores the domain like 'cal.com', 'acme.com'. We should use that itself for the redirect.

Reply:
> > we should take url input from them and redirect to that url. Similar to how we do it for successful booking redirect.
> 
> we should not take url input from them, cause this can lead to an abuse vector (adidas.cal.com => adidas.phishing-website.com) instead we should redirect to their verified Org domain (the one they use for all auto-invites for emails)
> 
> it's also one less input


Makes much more sense 👍